src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,473 @@
+import argparse
+import json
+import os
+import logging
+
+import boto3
+import datacompy
+import pandas as pd
+from pyarrow import fs
+import pyarrow.parquet as pq
+import synapseclient
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+INDEX_FIELD_MAP = {
+    "dataset_enrolledparticipants": ["ParticipantIdentifier"],
+    "dataset_fitbitprofiles": ["ParticipantIdentifier", "ModifiedDate"],
+    "dataset_fitbitdevices": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitactivitylogs": ["LogId"],
+    "dataset_fitbitdailydata": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitintradaycombined": ["ParticipantIdentifier", "Type", "DateTime"],
+    "dataset_fitbitrestingheartrates": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitsleeplogs": ["LogId"],
+    "dataset_healthkitv2characteristics": ["HealthKitCharacteristicKey"],
+    "dataset_healthkitv2samples": ["HealthKitSampleKey"],
+    "dataset_healthkitv2samples_deleted": ["HealthKitSampleKey"],
+    "dataset_healthkitv2heartbeat": ["HealthKitHeartbeatSampleKey"],
+    "dataset_healthkitv2statistics": ["HealthKitStatisticKey"],
+    "dataset_healthkitv2clinicalrecords": ["HealthKitClinicalRecordKey"],
+    "dataset_healthkitv2electrocardiogram": ["HealthKitECGSampleKey"],
+    "dataset_healthkitv2workouts": ["HealthKitWorkoutKey"],
+    "dataset_healthkitv2activitysummaries": ["HealthKitActivitySummaryKey"],
+    "dataset_googlefitsamples": ["GoogleFitSampleKey"],
+    "dataset_symptomlog": ["DataPointKey"],
+}
+
+
+def read_args():
+    parser = argparse.ArgumentParser(
+        description=(
+            "Compare parquet datasets between two namespaced S3 bucket locations"
+        )
+    )
+    parser.add_argument(
+        "--staging-namespace",
+        required=True,
+        type = validate_args,
+        help="The name of the staging namespace to use",
+        default="staging"
+    )
+    parser.add_argument(
+        "--main-namespace",
+        required=True,
+        type = validate_args,
+        help=("The name of the main namespace to use"),
+        default="main"
+    )
+    parser.add_argument(
+        "--parquet-bucket",
+        required=True,
+        type = validate_args,
+        help=("The name of the S3 bucket containing the S3 files to compare"),
+        default="recover-dev-processed_data"
+    )
+    args = parser.parse_args()
+    return args
+
+
+def validate_args(value : str) -> str:
+    """Checks to make sure none of the input command line arguments are empty strings
+
+    Args:
+        value (str): the value of the command line argument parsed by argparse
+
+    Raises:
+        argparse.ArgumentTypeError: when value is an empty string
+
+    Returns:
+        str: the value as is
+    """
+    if value == "":
+        raise argparse.ArgumentTypeError('Argument value cannot be an empty string')
+    return value
+
+
+def get_s3_file_key_for_comparison_results(
```

Comment:
Are we alright with this format here when saving the comparison reports?

Reply:
Sounds good, just noting here only quirk I noticed about its behavior is that it's not happy with slashes if you have something like:

-  `os.path.join("s3://", "some_bucket_name", "some_folder", "/")` which turns the entire path into `/`
- `os.path.join("s3://", "/some_bucket_name", "some_folder")` which turns the path into `/some_bucket_name/some_folder`

Reply:
`s3://` is not technically part of the path. That's the scheme of the URI. So you can do something like:
```
f"s3://{os.path.join(bucket, key_prefix)}"
```